src/v/archival/ntp_archiver_service.cc

```diff
@@ -2026,15 +2026,29 @@ ss::future<ntp_archiver::batch_result> ntp_archiver::wait_all_scheduled_uploads(
       .compacted_upload_result = compacted_result};
 }
 
+model::offset ntp_archiver::max_uploadable_offset() const {
+    // We impose an additional (LSO) constraint on the uploadable offset to
+    // as we need to have a complete index of aborted transactions if any
+    // before we can upload a segment.
+    return std::min(
+      model::prev_offset(_parent.last_stable_offset()),
+      _parent.committed_offset());
+}
+
 ss::future<ntp_archiver::batch_result> ntp_archiver::upload_next_candidates(
   std::optional<model::offset> lso_override) {
```

Comment:
the code path that sets `lso_override` should also be affected by this

Reply:
What do you mean by "affected"? Anything beside tests use it? Seems like only tests use it and they are not affected.

Separately, should we clamp it to committed offset too just in case someone tries to use it in the future inside real code?

Reply:
I thought that we're using it to make timeboxed uploads but no. We're only using it in tests.